src/support/err.c

```diff
@@ -504,12 +502,11 @@ __wt_illegal_value(WT_SESSION_IMPL *session, const char *name)
 	    name == NULL ? "" : name, name == NULL ? "" : ": ",
 	    "encountered an illegal file format or internal value");
 
-#if !defined(HAVE_DIAGNOSTIC)
-	return (__wt_panic(session));
+#if defined(HAVE_DIAGNOSTIC)
+	__wt_abort(session);                    /* Drop core if testing. */
 #endif
 
-	__wt_abort(session);			/* Drop core if testing. */
-	/* NOTREACHED */
+	return (WT_ERROR);
```

Comment:
`__wt_illegal_value` is only called when we've hit a truly fatal error (for example, the default case in a switch statement that "can't happen", or we failed checksum on a block we read), it should not be hit by salvage or verify (under normal conditions).

In short, panic is the right behavior for WiredTiger internally, we can't proceed further.

@daveh86 could change the name from `__wt_illegal_value` to `__wt_panic_something`, to clarify that?

Not sure what to suggest about MongoDB: could it push out the error message before it shuts down?